### PR TITLE
add energy sensor to pzem004t

### PR DESCRIPTION
**Pull request by AndreyHeaven**
## Description:
add read energy value from uart

sorry no PR for docs

i don't often do PR. please point me to problem in PR or code if its exist 

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

**Comment by glmnet** [comment]
Looks good, can you add 

````yaml
    energy:
      name: "PZEM004T Energy"
````

to test3.yaml?, 

Also adding docs for this is easy you can use "Edit this page in GitHub" at bottom of [docs](https://esphome.io/components/sensor/pzem004t.html) if you don't want to setup docs enverionment.

**Comment by AndreyHeaven** [comment]
> Looks good, can you add
> 
> ```yaml
>     energy:
>       name: "PZEM004T Energy"
> ```
> 
> to test3.yaml?,
> 
> Also adding docs for this is easy you can use "Edit this page in GitHub" at bottom of [docs](https://esphome.io/components/sensor/pzem004t.html) if you don't want to setup docs enverionment.

done

File: esphome/const.py [hunk 1]
```diff
@@ -562,6 +563,7 @@
 UNIT_VOLT_AMPS = 'VA'
 UNIT_VOLT_AMPS_REACTIVE = 'VAR'
 UNIT_WATT = 'W'
+UNIT_WATT_PER_HOUR = 'Wh'
```
Comment (OttoWinter) [inline esphome/const.py]: Watt per hour != Wh

Wh is Watthour, Watt per hour is a very different thing (and would be written as W/h)

**Review by OttoWinter — COMMENTED** [review]
Hi, thanks for adding this!

However, I don't think it's very useful in this form - the energy field will just keep on incrementing forever and there's no way to reset it back to zero. Or do I misunderstand something here (maybe the PZEM automatically resets, but I don't think it has a clock source).

Please also see https://github.com/esphome/feature-requests/issues/49#issuecomment-545993308

**Comment by sendorm** [comment]
Actually this is quite useful as is. You use the values with the `utility meter` home assistant component. Not resetting is not relevant as the component resets automatically in defined intervals. 

Relevant setup:
```
utility_meter:
  energy:
    source: sensor.energy_kwh
    cycle: monthly
  daily_energy:
    source: sensor.energy_kwh
    cycle: daily

**Comment by OttoWinter** [comment]
@sendorm Ok, I understand it can be used without resetting then.

Also, this PR needs an update to docs as well - and it would be good to add a note about `utility_meter` there too.

**Comment by AndreyHeaven** [comment]
add PR for docs esphome/esphome-docs#430

**Pull request closed without merging**